File: OneSignalSDK/onesignal/src/main/java/com/onesignal/outcomes/OSOutcomeEventsClient.java
package com.onesignal.outcomes;

import com.onesignal.OneSignalAPIClient;
import com.onesignal.OneSignalApiResponseHandler;
import com.onesignal.outcomes.domain.OutcomeEventsService;

import org.json.JSONObject;

abstract class OSOutcomeEventsClient implements OutcomeEventsService {

    final OneSignalAPIClient client;

    OSOutcomeEventsClient(OneSignalAPIClient client) {
        this.client = client;
    }

    public abstract void sendOutcomeEvent(JSONObject object, OneSignalApiResponseHandler responseHandler);

}

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/outcomes/data/OSOutcomeEventsClient.kt
package com.onesignal.outcomes.data

import com.onesignal.OneSignalAPIClient
import com.onesignal.OneSignalApiResponseHandler
import org.json.JSONObject

internal abstract class OSOutcomeEventsClient(val client: OneSignalAPIClient) : OutcomeEventsService {
    abstract override fun sendOutcomeEvent(jsonObject: JSONObject, responseHandler: OneSignalApiResponseHandler)
}